wrappers/python/simtk/openmm/app/internal/amber_file_parser.py

```diff
@@ -1070,7 +987,29 @@ def readAmberSystem(topology, prmtop_filename=None, prmtop_loader=None, shake=No
         elif gbmodel == 'GBn2':
             gb = customgb.GBSAGBn2Force(solventDielectric, soluteDielectric, 'ACE', cutoff, implicitSolventKappa)
         else:
-            raise Exception("Illegal value specified for implicit solvent model")
+            raise ValueError("Illegal value specified for implicit solvent model")
+        if isinstance(gb, mm.GBSAOBCForce):
+            # Built-in GBSAOBCForce does not have getStandardParameters, so use
+            # the one from the equivalent CustomGBForce
+            gb_parms = customgb.GBSAOBC2Force.getStandardParameters(topology)
+        else:
+            gb_parms = type(gb).getStandardParameters(topology)
+        # Replace radii and screen, but screen *only* gets replaced by the
+        # prmtop contents for HCT, OBC1, and OBC2. GBn and GBn2 both override
+        # the prmtop screen factors from LEaP in sander and pmemd
+        if gbmodel in ('HCT', 'OBC1', 'OBC2'):
+            screen = [float(s) for s in prmtop._raw_data['SCREEN']]
+        else:
+            screen = [gb_parm[1] for gb_parm in gb_parms]
+        radii = [float(r)/10 for r in prmtop._raw_data['RADII']]
+        warned = False
+        for i, (r, s) in enumerate(zip(radii, screen)):
+            if abs(r - gb_parms[i][0]) > 1e-4 or abs(s - gb_parms[i][1]) > 1e-4:
+                if not warned:
+                    warnings.warn('Non-optimal GB parameters detected for GB '
+                                  'model %s' % gbmodel)
+                    warned = True
```

Comment:
Each GB model has associated with it a "suggested" set of intrinsic radii for which the parameters were derived.  Each of the `CustomAmberGBForce` objects know what these parameters are.  However, the Amber prmtop file stores these parameters (which are assigned by LEaP).

If these disagree, that means that the user did not specify the *correct* radii that were used in the parametrization inside tleap.

I forget which issue this was discussed, but @jchodera raised concerns about always using the *correct* GB radii for Amber topologies, since that's a marked change in behavior of what used to be done.  I would personally rather always use the suggested parameters, but in the interest of maintaining backwards compatibility and not having results change from version to version, I kept the "old" behavior and was verbose with a warning if the parameters differed from "ideal" in the prmtop for the requested GB model.